src/main/java/ch/admin/bag/covidcertificate/service/CovidCertificateDtoMapperService.java

```diff
@@ -59,13 +82,13 @@ public VaccinationTouristCertificatePdf toVaccinationTouristCertificatePdf(Vacci
     }
 
     public TestCertificateQrCode toTestCertificateQrCode(TestCertificateCreateDto createDto) {
-        var testValueSet = valueSetsService.getIssuableTestDto(createDto.getTestInfo().get(0));
+        var testValueSet = valueSetsService.validateAndGetIssuableTestDto(createDto.getTestInfo().get(0).getTypeCode(), createDto.getTestInfo().get(0).getManufacturerCode());
         return TestCertificateQrCodeMapper.toTestCertificateQrCode(createDto, testValueSet);
 
     }
 
     public TestCertificatePdf toTestCertificatePdf(TestCertificateCreateDto createDto, TestCertificateQrCode qrCodeData) {
-        var testValueSet = valueSetsService.getIssuableTestDto(createDto.getTestInfo().get(0));
+        var testValueSet = valueSetsService.validateAndGetIssuableTestDto(createDto.getTestInfo().get(0).getTypeCode(), createDto.getTestInfo().get(0).getManufacturerCode());
```

Comment:
Repeatedly using "createDto.getTestInfo().get(0)" should be replaced by extraction to value and referencing, imho

Reply:
I have applied this suggestion and refactored the affected code in this class.